### ISSUE TITLE
[Feature request] Finding all images in nested directories

### ISSUE THREAD

**Issue by gurgalex** [Davejkane/riv#49]
# Problem

Unable to view files existing at multiple levels in a nested directory.

# Desired Behavior
Viewing of all the images listed.
```
# File tree of Pictures folder
image.png
dir1/image.png
dir1/depth2/image.png
dir1/depth2_2/image.png
```

# Syntax

Is there a syntax to express that all images of certain types should be searched for up to a certain directory depth?

pseudo syntax
`riv -f Pictures/**[*.png]/**[*.png]/*.png`

# Expected behavior
Find all images of type png in base Pictures folder.
Inside any directories in `Pictures`, find any png files.
Inside any sub-sub directories of Pictures find any png files.

Thoughts?

**Comment by Davejkane** [comment]
I'm not sure I understand. I'm pretty sure on zsh if I do `riv **/*.png` I get this behaviour. As for limiting the depth, this probably something that should be done by the shell. At runtime it's a different story, and I'll see if I can take it into account in [#40](https://github.com/Davejkane/riv/issues/40). I'm not sure how to do this in bash.

**Comment by gurgalex** [comment]
Tested on zsh and it has the expected behavior.  (4 of 4 images)

Bash shell only showed.

```
dir1/image.png
```

Closing as it is shell specific.

**Comment by gurgalex** [comment]
Hmm, may be a reason to bring back quoting globs.

`riv "**/*.png"` would have the same result regardless of shell expansion support (plus if a user has say 10k files and possibly reaches the max supported cli arguments for a program).

https://docs.rs/shellexpand/1.0.0/

**Comment by gurgalex** [comment]
Closed in favor of #50

**Comment by Davejkane** [comment]
Yeah, I think I agree with you on this. Especially after doing a little bit of reading about bash. Also, I'll need this to take care of [#40](https://github.com/Davejkane/riv/issues/40) anyway. Please close this issue anyway and I'll take care of it in that branch. Although I think it's going to be a nightmare coz I don't think that library you've linked does glob expansions.
